tiledb/sm/c_api/tiledb.cc

```diff
@@ -5158,6 +5170,70 @@ int32_t tiledb_deserialize_query_est_result_sizes(
   return TILEDB_OK;
 }
 
+int32_t tiledb_serialize_config(
+    tiledb_ctx_t* ctx,
+    const tiledb_config_t* config,
+    tiledb_serialization_type_t serialize_type,
+    int32_t client_side,
+    tiledb_buffer_t** buffer) {
+  // Sanity check
+  if (sanity_check(ctx) == TILEDB_ERR ||
+      sanity_check(ctx, config) == TILEDB_ERR)
+    return TILEDB_ERR;
+
+  // Create buffer
+  if (tiledb_buffer_alloc(ctx, buffer) != TILEDB_OK ||
+      sanity_check(ctx, *buffer) == TILEDB_ERR)
+    return TILEDB_ERR;
+
+  if (SAVE_ERROR_CATCH(
+          ctx,
+          tiledb::sm::serialization::config_serialize(
+              config->config_,
+              (tiledb::sm::SerializationType)serialize_type,
+              (*buffer)->buffer_,
+              client_side)))
```

Comment:
we might have this pattern elsewhere, but i think `buffer` leaks in this error path

Reply:
The user is responsible for creating/deallocated the tiledb_buffer_t, as such I don't think there is leak.

Reply:
Only for an `TILEDB_OK` status. Similar to L5230

Reply:
All the serialized APIs work the same here, the user is responsible for checking if `*buffer != nullptr` and freeing if so. Do you want to change the behavior for everything to self free on error?